NuRadioReco/modules/io/NuRadioRecoio.py

```diff
@@ -368,6 +378,7 @@ def get_detector(self):
                         source='dictionary', dictionary=detector_dict,
                         default_station=detector_dict.get('default_station', None),
                         default_channel=detector_dict.get('default_channel', None),
+                        assume_inf=assume_inf, antenna_by_depth=antenna_by_depth,
```

Comment:
instead of the code block from 360-369 you can only write `assume_inf=detector_dict['detector_parameters'].get('assume_inf', None)`

Reply:
ah, in the old files the nested dict does not exist neither. In that case I would get rid of the nested dict to simplify the code.

Reply:
With the change in NuradioRecoio.py, the 'detector_parameters' key will always exist. If it's not present in the .nur file, it will just be an empty dictionary, so your solution works perfectly 👍🏽